### ISSUE TITLE
pl.from_arrow excludes non-numeric columns when reading empty table

### ISSUE THREAD

**Issue by johanbog** [pola-rs/polars#2754]
#### What language are you using?

Python

#### Which feature gates did you use?

This can be ignored by Python & JS users.

#### What version of polars are you using?

0.13.5

#### What operating system are you using polars on?

macOS 12.1

#### What language version are you using
python 3.9.10, pyarrow 7.0.0

#### Describe your bug.

Creating DataFrame from an empty arrow table removes all non-numeric columns

#### What are the steps to reproduce the behavior?
```
import pandas as pd
import polars as pl
import pyarrow

df = pd.DataFrame(columns=['a', 'b', 'c'])
df['a'] = df['a'].astype(str)
df['b'] = df['b'].astype(float)
df['c'] = df['c'].astype(object)

print(pl.from_arrow(pyarrow.Table.from_pandas(df)))
```

Related to https://github.com/pola-rs/polars/issues/2705

#### What is the actual behavior?

```
shape: (0, 1)
┌─────┐
│ b   │
│ --- │
│ f64 │
╞═════╡
└─────┘
```

#### What is the expected behavior?

Not deleted the non-numeric columns

**Comment by ghuls** [comment]
Also related to #2749 I think as the `str` and `object` column generate a NullArray.
```python
In [110]: pyarrow.Table.from_pandas(df)
Out[110]:
pyarrow.Table
a: null
b: double
c: null
__index_level_0__: null
----
a: [0 nulls]
b: [[]]
c: [0 nulls]
__index_level_0__: [0 nulls]
```

**Comment by ritchie46** [comment]
Where does type information go then? Can we prevent this?

**Comment by ghuls** [comment]
pyarrow saves some extra info for restoring to pandas:
```python
In [121]: pyarrow.Table.from_pandas(df).schema.metadata
Out[121]: {b'pandas': b'{"index_columns": ["__index_level_0__"], "column_indexes": [{"name": null, "field_name": null, "pandas_type": "unicode", "numpy_type": "object", "metadata": {"encoding": "UTF-8"}}], "columns": [{"name": "a", "field_name": "a", "pandas_type": "empty", "numpy_type": "object", "metadata": null}, {"name": "b", "field_name": "b", "pandas_type": "float64", "numpy_type": "float64", "metadata": null}, {"name": "c", "field_name": "c", "pandas_type": "empty", "numpy_type": "object", "metadata": null}, {"name": null, "field_name": "__index_level_0__", "pandas_type": "empty", "numpy_type": "object", "metadata": null}], "creator": {"library": "pyarrow", "version": "7.0.0"}, "pandas_version": "1.4.1"}'}
```

**Comment by ghuls** [comment]
Or in a more formatted way:
```python
In [143]: pyarrow.Table.from_pandas(df).schema.pandas_metadata
Out[143]:
{'index_columns': ['__index_level_0__'],
 'column_indexes': [{'name': None,
   'field_name': None,
   'pandas_type': 'unicode',
   'numpy_type': 'object',
   'metadata': {'encoding': 'UTF-8'}}],
 'columns': [{'name': 'a',
   'field_name': 'a',
   'pandas_type': 'empty',
   'numpy_type': 'object',
   'metadata': None},
  {'name': 'b',
   'field_name': 'b',
   'pandas_type': 'float64',
   'numpy_type': 'float64',
   'metadata': None},
  {'name': 'c',
   'field_name': 'c',
   'pandas_type': 'empty',
   'numpy_type': 'object',
   'metadata': None},
  {'name': None,
   'field_name': '__index_level_0__',
   'pandas_type': 'empty',
   'numpy_type': 'object',
   'metadata': None}],
 'creator': {'library': 'pyarrow', 'version': '7.0.0'},
 'pandas_version': '1.4.1'}
```

**Comment by ghuls** [comment]
It looks like for Null arrays, pyarrow returns np.object_

```python
In [212]: pa.null().to_pandas_dtype()
Out[212]: numpy.object_
```
https://issues.apache.org/jira/browse/ARROW-8442

```python
import pandas as pd
import polars as pl
import pyarrow as pa

df = pd.DataFrame(columns=['a', 'b', 'c'])
df['a'] = df['a'].astype(str)
df['b'] = df['b'].astype(float)
df['c'] = df['c'].astype(object)
df['d'] = df['d'].astype(int)


In [208]: df
Out[208]:
Empty DataFrame
Columns: [a, b, c, d]
Index: []

In [209]: df.dtypes
Out[209]:
a     object
b    float64
c     object
d      int64
dtype: object

In [210]: pa.Table.from_pandas(df)
Out[210]:
pyarrow.Table
a: null
b: double
c: null
d: int64
__index_level_0__: null
----
a: [0 nulls]
b: [[]]
c: [0 nulls]
d: [[]]
__index_level_0__: [0 nulls]

In [211]: pl.DataFrame(pa.Table.from_pandas(df))
Out[211]:
shape: (0, 2)
┌─────┬─────┐
│ b   ┆ d   │
│ --- ┆ --- │
│ f64 ┆ i64 │
╞═════╪═════╡
└─────┴─────┘

```

**Comment by ritchie46** [comment]
fixed by #2757
